File: ios/Modules/Sources/Auth/api/SecureStore.swift
import Foundation

/// Key/value store with OS-level encryption at rest.
///
/// Holds small secrets only (JWTs, seed phrase, per-device keys). Not
/// for bulk data — that goes to GRDB (Wave 7). Keeping the surface
/// narrow lets us swap the Keychain-backed impl for a hardware-enclave
/// one without any call-site churn.
public protocol SecureStore: Sendable {
    func getString(_ key: String) -> String?
    func setString(_ key: String, _ value: String?)
    func clear()
}

File: android/app/src/main/java/sol/vortexx/android/auth/api/SecureStore.kt
package sol.vortexx.android.auth.api

/**
 * Key/value store with OS-level encryption at rest.
 *
 * Responsibility: hold small secrets (JWTs, seed phrase, device keys).
 * NOT for arbitrary data — that goes to DataStore or Room. Keeping the
 * surface this narrow means consumers can't abuse it for bulk storage
 * and we can swap the backing Keystore-wrapped SharedPreferences for
 * e.g. AndroidX EncryptedFile in a later wave with no call-site churn.
 */
interface SecureStore {
    fun getString(key: String): String?
    fun putString(key: String, value: String?)
    fun clear()
}
